clang/lib/CodeGen/CGCall.cpp

```diff
@@ -2408,6 +2408,12 @@ void CodeGenModule::ConstructAttributeList(StringRef Name,
           FuncAttrs.addAttribute(llvm::Attribute::NoReturn);
         NBA = Fn->getAttr<NoBuiltinAttr>();
       }
+
+      for (const FunctionEffectWithCondition &Fe : Fn->getFunctionEffects()) {
```

Comment:
I think I remember Vitaly requesting unit tests for this stuff - it'd be good to investigate how difficult that would be for this!

Reply:
Check these tests out in clang/test/Driver